quil-rs/src/program/mod.rs

```diff
@@ -379,6 +381,74 @@ impl Program {
         Ok(expanded_program)
     }
 
+    /// Return a copy of the [`Program`] wrapped in a loop that repeats `iterations` times.
+    ///
+    /// The loop is constructed by wrapping the body of the program in classical Quil instructions.
+    /// The given `loop_count_reference` must refer to an INTEGER memory region. The value at the
+    /// reference given will be set to `iterations` and decremented in the loop. The loop will
+    /// terminate when the reference reaches 0. For this reason your program should not itself
+    /// modify the value at the reference unless you intend to modify the remaining number of
+    /// iterations (i.e. to break the loop).
+    ///
+    /// The loop is constructed with [`TargetPlaceholder`]s in order to prevent conflicts
+    /// with existing label names in the program. These placeholders are resolved before
+    /// returning the program.
+    ///
+    /// If `iterations` is 0, then a copy of the program is returned without any changes.
+    pub fn wrap_in_loop(&self, loop_count_reference: MemoryReference, iterations: u32) -> Self {
+        if iterations == 0 {
+            return self.clone();
+        }
+
+        let mut looped_program = self.clone_without_body_instructions();
+        let start_target = TargetPlaceholder::new("__loop-start".to_string());
+        let end_target = TargetPlaceholder::new("__loop-end".to_string());
```

Comment:
Do these need to be unique at all? What happens if you call this multiple times e.g. `wrap_in_loop().wrap_in_loop()` or otherwise have nesting?

Reply:
Yeah, `pdoc` uses a markdown extension that makes it such that paths wrapped in single backticks links to that path in the documentation. Double backticks is normal inline code.

Not sure why the action hasn't updated the deployment notification on this PR, but [you can see how it renders in the preview here](https://rigetti.github.io/quil-rs/pr-preview/pr-321/quil/program.html#Program.wrap_in_loop).

Reply:
By default, the placeholders resolve by checking for conflicting targets and appending a unique integer to the end, so I believe that should have worked as expected.

Though, that is a moot point now that this method makes the user provide `Target`s per Kalan's comment.